slider/beatmap.py

```diff
@@ -1680,6 +1680,58 @@ def _resolve_stacking_old(self, hit_objects, ar, cs):
 
         return hit_objects
 
+    @lazyval
+    def _hit_object_times(self):
+        """a (sorted) list of hitobject time's, so they can be searched with
+        ``np.searchsorted``
+        """
+        return [hitobj.time for hitobj in self._hit_objects]
+
+    def closest_hitobject(self, t, side="left"):
+        """The hitobject closest in time to ``t``.
+
+        Parameters
+        ----------
+        t : :class:`datetime.timedelta`
+            The time to find the hitobject closest to.
+        side : {"left", "right"}
+            Whether to prefer the earlier (left) or later (right) hitobject
+            when breaking ties.
+
+        Returns
+        -------
+        hit_object : :class:`~.HitObject`
+            The closest hitobject in time to ``t``.
+        None
+            If the beatmap has no hitobjects.
+        """
+        # if the beatmap only has one object, return that, else return None
+        if len(self._hit_objects) <= 1:
+            return self._hit_objects[0] if self._hit_objects else None
```

Comment:
I think this should throw some kind of exception instead of returning None. Users can always catch this, but most use cases don't want to accidentally pass `None` around.